### ISSUE TITLE
telepresence auto exit

### ISSUE THREAD

**Issue by Crow0687** [telepresenceio/telepresence#2830]
Excuse me, when I run telepresence, I automatically exit

**1. run:** 
sudo telepresence --expose 8080:80 --also-proxy 172.10.0.0/16 --namespace regulation-qa

**output:**
T: Warning: Telepresence has only been testing on version 1.* clusters
T: Warning: kubectl 1.25.0 may not work correctly with cluster version 4.5.7 due to the version discrepancy. See https://kubernetes.io/docs/setup/version-skew-policy/ for more information.

T: Using a Pod instead of a Deployment for the Telepresence proxy. If you experience problems, please file an issue!
T: Set the environment variable TELEPRESENCE_USE_DEPLOYMENT to any non-empty value to force the old behavior, e.g.,
T:     env TELEPRESENCE_USE_DEPLOYMENT=1 telepresence --run curl hello

T: Starting proxy with method 'vpn-tcp', which has the following limitations: All processes are affected, only one telepresence can run per machine, and you can't use other VPNs. You may need to add cloud hosts and headless services
T:  with --also-proxy. For a full list of method limitations see https://telepresence.io/reference/methods.html
T: Volumes are rooted at $TELEPRESENCE_ROOT. See https://telepresence.io/howto/volumes.html for details.
T: Starting network proxy to cluster using new Pod telepresence-1666685500-841007-30791
T: Forwarding remote port 80 to local port 8080.

T: Mounting remote volumes failed, they will be unavailable in this session. If you are running on Windows Subystem for Linux then see https://github.com/datawire/telepresence/issues/115, otherwise please report a bug, attaching
T: telepresence.log to the bug report: https://github.com/datawire/telepresence/issues/new
T: Connected. Flushing DNS cache.
T: Setup complete. Launching your command.
T: Your process exited with return code -9.
T: Exit cleanup in progress
T: Cleaning up Service, Pod

**pod logs:**
kubectl --kubeconfig=/Users/yfmeng/Library/Caches/JetBrains/IntelliJIdea2022.2/kubernetes/regulation/api-provider/config665 logs telepresence-1666685500-841007-30791 --container=telepresence --namespace=regulation-qa --follow=true
Retrieving this pod's namespace from the process environment
Failed: TELEPRESENCE_CONTAINER_NAMESPACE not set
Reading this pod's namespace from the k8s service account
Pod's namespace is 'regulation-qa'
Listening...
2022-10-25T08:11:50+0000 [-] Loading ./forwarder.py...
2022-10-25T08:11:51+0000 [-] /etc/resolv.conf changed, reparsing
2022-10-25T08:11:51+0000 [-] Resolver added ('172.254.0.2', 53) to server list
2022-10-25T08:11:51+0000 [-] SOCKSv5Factory starting on 9050
2022-10-25T08:11:51+0000 [socks.SOCKSv5Factory#info] Starting factory <socks.SOCKSv5Factory object at 0x7f800e38cba8>
2022-10-25T08:11:51+0000 [-] DNSDatagramProtocol starting on 9053
2022-10-25T08:11:51+0000 [-] Starting protocol <twisted.names.dns.DNSDatagramProtocol object at 0x7f800e38ce80>
2022-10-25T08:11:51+0000 [-] Loaded.
2022-10-25T08:11:51+0000 [twisted.scripts._twistd_unix.UnixAppLogger#info] twistd 20.3.0 (/usr/bin/python3.6 3.6.8) starting up.
2022-10-25T08:11:51+0000 [twisted.scripts._twistd_unix.UnixAppLogger#info] reactor class: twisted.internet.epollreactor.EPollReactor.
2022-10-25T08:11:58+0000 [stdout#info] Sanity check: b'hellotelepresence-4.a.sanity.check.telepresence.io'
2022-10-25T08:11:58+0000 [stdout#info] Set DNS suffix we filter out to: [()]
2022-10-25T08:11:58+0000 [stdout#info] Result for b'hellotelepresence-5' is ['127.0.0.1']
2022-10-25T08:11:58+0000 [stdout#info] Sanity check: b'hellotelepresence-5.a.sanity.check.telepresence.io'
2022-10-25T08:11:59+0000 [stdout#info] Result for b'hellotelepresence-6' is ['127.0.0.1']
2022-10-25T08:11:59+0000 [stdout#info] Sanity check: b'hellotelepresence-6.a.sanity.check.telepresence.io'
2022-10-25T08:11:59+0000 [stdout#info] Result for b'hellotelepresence-7' is ['127.0.0.1']
2022-10-25T08:11:59+0000 [stdout#info] 12 query: b'lb._dns-sd._udp.0.100.168.192.in-addr.arpa'
2022-10-25T08:11:59+0000 [DNSDatagramProtocol (UDP)] DNSDatagramProtocol starting on 4892
2022-10-25T08:11:59+0000 [DNSDatagramProtocol (UDP)] Starting protocol <twisted.names.dns.DNSDatagramProtocol object at 0x7f800e317748>
2022-10-25T08:11:59+0000 [-] (UDP Port 4892 Closed)
2022-10-25T08:11:59+0000 [-] Stopping protocol <twisted.names.dns.DNSDatagramProtocol object at 0x7f800e317748>
2022-10-25T08:12:00+0000 [stdout#info] 12 query: b'54.100.168.192.in-addr.arpa'
2022-10-25T08:12:00+0000 [DNSDatagramProtocol (UDP)] DNSDatagramProtocol starting on 9776
2022-10-25T08:12:00+0000 [DNSDatagramProtocol (UDP)] Starting protocol <twisted.names.dns.DNSDatagramProtocol object at 0x7f800e317908>
2022-10-25T08:12:00+0000 [-] (UDP Port 9776 Closed)
2022-10-25T08:12:00+0000 [-] Stopping protocol <twisted.names.dns.DNSDatagramProtocol object at 0x7f800e317908>
2022-10-25T08:12:22+0000 [Poll#error] Failed to contact Telepresence client:
2022-10-25T08:12:22+0000 [Poll#error] Connection was refused by other side: 111: Connection refused.
2022-10-25T08:12:22+0000 [Poll#warn] Perhaps it's time to exit?
rpc error: code = Unknown desc = Error: No such container: cbe1d00e69f4824e480bdc27f7026b72a5cba80d9324120d33c2cbd69c88b210
Process finished with exit code 0

[telepresence.log](https://github.com/telepresenceio/telepresence/files/9858419/telepresence.log)

**telepresence version:** 0.109

**kubectl version**
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.0", GitCommit:"a866cbe2e5bbaa01cfd5e969aa3e033f3282a8a2", GitTreeState:"clean", BuildDate:"2022-08-23T17:44:59Z", GoVersion:"go1.19", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:45:25Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}

**Comment by clodo** [comment]
I have a very similar problem after updating to the latest version of mac "ventura"

T: Mounting remote volumes failed, they will be unavailable in this session. If you are running on Windows Subystem for Linux then see
T: https://github.com/datawire/telepresence/issues/115, otherwise please report a bug, attaching telepresence.log to the bug report:
T: https://github.com/datawire/telepresence/issues/new
T: Connected. Flushing DNS cache.
T: Setup complete. Launching your command.
T: Your process exited with return code -9.
T: Exit cleanup in progress

**Comment by aaccioly** [comment]
Same problem here. Since I've upgraded to macOS Ventura the telepresence process exits with code -9. Any known workarounds?

**Comment by cindymullins-dw** [comment]
Legacy Telepresence (v1) is no longer actively supported so am closing this issue.  Please upgrade to Telepresence - you can refer to the [quick start here](https://www.getambassador.io/docs/telepresence/latest/quick-start/).  If issues persist on that version please let us know.

**Comment by Crow0687** [comment]
If your version of k8s is less than 1.24, you can try [kt](https://github.com/alibaba/kt-connect)

**Comment by oscarlgz** [comment]
@cindymullins-dw unfortunately v2 isn't a good fit for many use cases (services that periodically perform tasks, services that operate on job queues, etc). See issues [here](https://github.com/telepresenceio/telepresence/issues/1608) and [here](https://github.com/telepresenceio/telepresence/issues/1646). I'm hoping a new version of Telepresence will support these workloads eventually and bring it up to parity with v1 in this regard.

**Comment by vitgorbunov** [comment]
Any workaround? I also don't have an ability to use telepresence 2 yet.

**Comment by vitgorbunov** [comment]
It seems that I have found a workaround.
By default telepresence runs command `bash --norc` in the end and it fails with code "-9"
I've tried to launch it with `--run zsh -f` or `--run bash`, but go the same error code.

But when I append `--run env -i bash --norc` to the telepresence command, the process doesn't fail and everything works fine.
The resulting shell doesn't have env variables like `TELEPRESENCE_ROOT`, so I have to use `ls -l /tmp/ | grep tel-` and find the latest directory. Other than that it's enough for my use case.
